src/git/commit.rs

```diff
@@ -180,6 +233,52 @@ mod test {
         Ok(())
     }
 
+    #[sealed_test]
+    fn crate_signed_ssh_commit_ok() -> Result<()> {
+        init_builtin_logger();
+
+        // Arrange
+        let crate_dir = std::env::var("CARGO_MANIFEST_DIR")?;
+
+        // Source a ssh-agent and set env variables
+        let output = run_fun!(
+            ssh-agent;
+        )?;
+
+        let variables = output.split(';').map(|s| s.trim()).collect::<Vec<&str>>();
+        let ssh_auth_sock = variables[0].split('=').collect::<Vec<&str>>()[1];
+        let ssh_agent_pid = variables[2].split('=').collect::<Vec<&str>>()[1];
+        std::env::set_var("SSH_AUTH_SOCK", ssh_auth_sock);
+        std::env::set_var("SSH_AGENT_PID", ssh_agent_pid);
+
+        run_cmd!(
+            git init;
+            chmod 600 $crate_dir/tests/assets/sshkey;
+            chmod 600 $crate_dir/tests/assets/sshkey.pub;
+            git config --local user.signingkey "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIHsnHukmf4SX31jdbf+aZjH2pvmHwuz7ysxdjErMK+i2";
+            git config --local gpg.format ssh;
+            git config --local gpg.ssh.program ssh-keygen;
+            ssh-add $crate_dir/tests/assets/sshkey;
+            echo changes > file;
+            git add .;
+        )?;
+
+        let repo = Repository::open(".")?;
+
+        // Act
+        let oid = repo.commit("feat: a test commit", true);
```

Comment:
This now takes an additional boolean argument to allow for empty delta

Reply:
Ah i see, and i just realised this doesn't error on normal builds because its a test, im a little silly.